src/EndpointV2/Rule/TreeRule.php

```diff
@@ -0,0 +1,61 @@
+<?php
+
+namespace Aws\EndpointV2\Rule;
+
+use Aws\EndpointV2\Ruleset\RulesetStandardLibrary;
+
+class TreeRule extends Rule
+{
+    /** @var array */
+    private $rules;
+
+    public function __construct(array $spec)
```

Comment:
Spec feels like an awkward name for the variable.  Specification is what I assume you're going for, but what does the specification contain?  Maybe $ruleDefinition.